actions/websocket_actions.jsx

```diff
@@ -373,7 +373,50 @@ function handleChannelMemberUpdatedEvent(msg) {
     dispatch({type: ChannelTypes.RECEIVED_MY_CHANNEL_MEMBER, data: channelMember});
 }
 
-function handleNewPostEvent(msg) {
+export function debouncePostEvent(func, wait) {
+    let timeout;
+    let queue = [];
+    return function fx(...args) {
+        // Called when timeout triggered
+        const triggered = () => {
+            timeout = null;
+            if (queue.length > 0) {
+                const posts = {};
+                let channelId = '';
+                queue.map((queuedArgs) => {
+                    const post = JSON.parse(queuedArgs[0].data.post);
+                    posts[post.id] = post;
+                    channelId = post.channel_id;
+                    return null;
+                });
+                dispatch({
+                    type: PostTypes.RECEIVED_POSTS,
+                    data: {posts},
+                    channelId,
+                });
+            }
+            queue = [];
+        };
+        if (timeout) {
+            // If the timeout is going this is the second or further event so queue them up.
+            if (queue.push(args) > 200) {
+                // Don't run us out of memory, give up if the queue gets insane
+                queue = [];
```

Comment:
Could we do this without dumping the entire queue of posts when it becomes overloaded? It might be nice to only ignore any ones over the limit in that case

Reply:
This only happens when there are over 200 posts within 100ms of the previous post. Not really worried about not breaking the channel in that case.